middleware/http_cluster_test.go

```diff
@@ -0,0 +1,307 @@
+		},
+		"empty request cluster and non-empty server cluster give an error if soft validation disabled": {
+			header: func(r *http.Request) {
+				r.Header[clusterutil.ClusterVerificationLabelHeader] = []string{""}
+			},
+			serverCluster:      "cluster",
+			expectedLogs:       `level=warn msg="request with no cluster validation label" path=/Test/Me cluster_validation_label=cluster soft_validation=%v`,
+			expectedStatusCode: http.StatusNetworkAuthenticationRequired,
+			expectedErrorMsg:   `rejected request with empty cluster validation label - it should be "cluster"`,
+		},
+		"no request cluster and non-empty server cluster give an error if soft validation disabled": {
+			serverCluster:      "cluster",
+			expectedLogs:       `level=warn msg="request with no cluster validation label" path=/Test/Me cluster_validation_label=cluster soft_validation=%v`,
+			expectedStatusCode: http.StatusNetworkAuthenticationRequired,
+			expectedErrorMsg:   `rejected request with empty cluster validation label - it should be "cluster"`,
+		},
+		"if the incoming context contains more than one cluster labels an error is returned if soft validation disabled": {
+			header: func(r *http.Request) {
+				r.Header[clusterutil.ClusterVerificationLabelHeader] = []string{"cluster", "another-cluster"}
+			},
+			serverCluster:      "cluster",
+			expectedLogs:       `level=warn msg="detected error during cluster validation label extraction" path=/Test/Me cluster_validation_label=cluster soft_validation=%v err="request header should contain exactly 1 value for key \"X-Cluster\", but it contains [cluster another-cluster]"`,
+			expectedStatusCode: http.StatusNetworkAuthenticationRequired,
+			expectedErrorMsg:   `rejected request: request header should contain exactly 1 value for key "X-Cluster", but it contains [cluster another-cluster]`,
+		},
+	}
+	for testName, testCase := range testCases {
+		for _, softValidation := range []bool{false, true} {
+			t.Run(fmt.Sprintf("%s softValidation=%v", testName, softValidation), func(t *testing.T) {
+				defer func() {
+					r := recover()
+					require.Equal(t, testCase.shouldPanic, r != nil)
+				}()
+				buf := bytes.NewBuffer(nil)
+				logger := createLogger(t, buf)
+				m := ClusterValidationMiddleware(testCase.serverCluster, nil, softValidation, logger)
+				handler := Merge(m).Wrap(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusOK)
+				}))
+
+				recorder := httptest.NewRecorder()
+				req, err := http.NewRequest("GET", "/Test/Me", nil)
+				require.NoError(t, err)
+				if testCase.header != nil {
+					testCase.header(req)
+				}
+				handler.ServeHTTP(recorder, req)
+				if softValidation {
+					require.Equal(t, http.StatusOK, recorder.Code)
+				} else {
+					require.Equal(t, testCase.expectedStatusCode, recorder.Code)
+					if recorder.Code != http.StatusOK {
+						var clusterValidationErr clusterValidationError
```

Comment:
Could you also test that the server responds with `Content-Type` `application/json`?